massa-wallet/src/lib.rs

```diff
@@ -24,27 +28,39 @@ pub struct Wallet {
     pub keys: Map<Address, (PublicKey, PrivateKey)>,
     /// Path to the file containing the private keys (not encrypted)
     pub wallet_path: PathBuf,
+    /// Password
+    pub password: String,
 }
 
 impl Wallet {
     /// Generates a new wallet initialized with the provided file content
-    pub fn new(path: PathBuf) -> Result<Wallet, WalletError> {
-        let keys = if path.is_file() {
-            serde_json::from_str::<Vec<PrivateKey>>(&std::fs::read_to_string(&path)?)?
+    pub fn new(path: PathBuf, password: String) -> Result<Wallet, WalletError> {
+        let content = &std::fs::read(&path)?[..];
+        let priv_keys = if !content.is_empty() {
+            // If wallet contains data decipher it
+            let decrypted_content = decrypt(&password, content)?;
+            serde_json::from_slice::<Vec<PrivateKey>>(&decrypted_content[..])?
         } else {
-            Vec::new()
+            // If wallet is empty add a new private key
+            vec![generate_random_private_key()]
         };
-        let keys = keys
+        let keys = priv_keys
             .iter()
             .map(|key| {
                 let pub_key = derive_public_key(key);
                 Ok((Address::from_public_key(&pub_key), (pub_key, *key)))
             })
             .collect::<Result<Map<Address, _>, WalletError>>()?;
-        Ok(Wallet {
+        let wallet = Wallet {
             keys,
             wallet_path: path,
-        })
+            password,
+        };
+        if content.is_empty() {
```

Comment:
I don't get this part

Reply:
Sole purpose was the case where user does not create a key in the first time he launches the client, but I'm currently changing this to a new behaviour.